tests_e2e/pipeline/pipeline-cleanup.yml

```diff
@@ -1,58 +1,59 @@
 #
 # Pipeline for cleaning up any remaining Resource Groups generated by the Azure.WALinuxAgent pipeline.
 #
-# Deletes any resource groups that are more than a day old and contain string "lisa-WALinuxAgent-"
+# Deletes any resource groups that are older than 'older_than' and match the 'name_pattern' regular expression
 #
-schedules:
-  - cron: "0 */12 * * *" # Run twice a day (every 12 hours)
-    displayName: cleanup build
-    branches:
-      include:
-      - develop
-    always: true
 
-trigger:
-  - develop
+parameters:
+  - name: name_pattern
+    displayName: Regular expression to match the name of the resource groups to delete
+    type: string
+    default: lisa-WALinuxAgent-.*
 
-pr: none
+  - name: older_than
+    displayName: Delete resources older than (use the syntax of the "date -d" command)
+    type: string
+    default: 1 day ago
+
+  - name: service_connections
+    type: object
+    default:
+      - azuremanagement
+#
+# TODO: Enable these services connections once we create test pipelines for all Azure clouds
+#
+#      - azuremanagement.china
+#      - azuremanagement.government
 
 pool:
   vmImage: ubuntu-latest
 
-variables:
-  - name: azureConnection
-    value: 'azuremanagement'
-  - name: rgPrefix
-    value: 'lisa-WALinuxAgent-'
-
 steps:
+  - ${{ each service_connection in parameters.service_connections }}:
+    - task: AzureCLI@2
+      inputs:
+        azureSubscription: ${{ service_connection }}
+        scriptType: 'bash'
+        scriptLocation: 'inlineScript'
+        inlineScript: |
+          set -euxo pipefail
 
-  - task: AzureKeyVault@2
```

Comment:
This was copied from the initial prototype. The only secret we need is the subscription ID, which can be obtained using the CLI in the next task, so removing this.